src/message/message-parser.js

```diff
@@ -48,17 +47,14 @@ module.exports = class MessageParser {
    *                    data: <array of strings>
    *                  }
    */
-  static parse (message) {
-    const parsedMessages = []
+  static parse (message, callback) {
     const rawMessages = message.split(C.MESSAGE_SEPERATOR)
 
     for (let i = 0; i < rawMessages.length; i++) {
       if (rawMessages[i].length > 2) {
-        parsedMessages.push(this.parseMessage(rawMessages[i]))
+        callback(this.parseMessage(rawMessages[i]), message)
```

Comment:
should we let callback take an error, incase of something like where message is null or not text so deepstream doesn't crash when someone sends a non text frame?